File: week_1/problem_set_1/mario-more/mario.c
// .......#..#
// ......##..##
// .....###..###
// ....####..####
// ...#####..#####
// ..######..######
// .#######..#######
// ########..########

#include <cs50.h>
#include <stdio.h>

void middleSpace()
{
    for (int i = 0; i < 2; i++)
    {
        printf(" ");
    }
}

void hash(row)
{
    for (int i = 0; i < row; i++)
    {
        printf("#");
    }
}

void leftSpace(space_number)
{
    for (int i = 0; i < space_number; i++)
    {
        printf(" ");
    }
}


int main(void)
{
    int size = 0;

    do
    {
        size = get_int("Height: ");
    }
    while (size <= 0 || size >= 9);

    int spaces = size - 1;

    for (int i = 1; i <= size; i++)
    {
        leftSpace(spaces);
        hash(i);
        middleSpace();
        hash(i);
        printf("\n");
        spaces = spaces - 1;
    }

}

File: week_06/sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    height = get_height()
    for i in range(height):
        print(left_side(i, height) + middle_space() + right_side(i))


def get_height():
    while True:
        height = get_int("Height: ")
        if height <= 0:
            print("Invalid input! Please type another number")
        elif height > 8:
            print("Invalid input! Please type another number")
        else:
            return height


def left_side(i, height):
    i = i + 1
    spaces = (height - i) * " "
    hashes = i * "#"
    return spaces + hashes


def middle_space():
    return "  "


def right_side(i):
    i = i + 1
    hashes = i * "#"
    return hashes


main()